File: codechef/Easy/Chef and Notebooks/Chef and Notebooks.py
for _ in range(int(input())):
    x,y,k,n = map(int,input().split())
    
    flag = 0
    rem = x - y
    for i in range(n):
        no_of_pages,price = map(int,input().split())
        if( no_of_pages >= rem and price<= k ):
            print("LuckyChef")
            flag = 1        
    if(flag == 0):
        print("UnluckyChef")

File: codechef/Easy/Chef and Notebooks/Chef and Notebooks.cpp
#include<iostream>
using namespace std;
int main (){
	int t;
	scanf("%d",&t);
	while(t--){
		int x,y,k,n,flag = 0 ;
	    scanf("%d%d%d%d",&x,&y,&k,&n);
		int nop, price;
		while(n--){
    	    scanf("%d%d",&nop,&price);
			if(nop>=(x-y) && price<=k ){
				flag = 1;
			}
		}
		if(flag==0)
		    cout<<"UnluckyChef"<<endl;
		else
		    cout<<"LuckyChef"<<endl;
	}
	return 0;
}
